ee/billing/quota_limiting.py

```diff
@@ -0,0 +1,31 @@
+from datetime import timedelta
+from enum import Enum
+from typing import List, Mapping
+
+from django.utils import timezone
+
+from posthog.cache_utils import cache_for
+from posthog.redis import get_client
+
+RATE_LIMITER_CACHE_KEY = "@posthog/quota-limits/"
+
+
+class QuotaResource(Enum):
+    EVENTS = "events"
+    RECORDINGS = "recordings"
+
+
+def replace_limited_teams(resource: QuotaResource, tokens: Mapping[str, int]) -> None:
+    pipe = get_client().pipeline()
+    pipe.zremrangebyscore(f"{RATE_LIMITER_CACHE_KEY}{resource.value}", "-inf", "+inf")
+    if tokens:
+        pipe.zadd(f"{RATE_LIMITER_CACHE_KEY}{resource.value}", tokens)  # type: ignore # (zadd takes a Mapping[str, int] but the derived Union type is wrong)
+    pipe.execute()
+
+
+@cache_for(timedelta(seconds=30))
```

Comment:
I would like to improve on this and have it return without the call to redis, and instead doing a background thread to refresh the data if the TTL is passed.

In addition we could then preload this value as part of the "liveness" work so that we always have a list in memory and are never slowing down the capture call.